lib/irb.rb

```diff
@@ -775,16 +779,25 @@ def signal_status(status)
       end
     end
 
+    def truncate_prompt_main(str) # :nodoc:
+      str = str.tr(CONTROL_CHARACTERS_PATTERN, ' ')
```

Comment:
It seems better not to use byte strings because they fail in ASCII-incompatible character encodings. I'll think about how to do it. 🤔 

```
irb(main):001:0> cws ("a".encode(Encoding::UTF_16BE))
irb: warn: can't alias delete from irb_delete.
=> "a"
/Users/mi/ghq/github.com/ruby/irb/lib/irb.rb:783:in `tr': incompatible character encodings: UTF-16BE and UTF-8 (Encoding::CompatibilityError)
```

Reply:
I thought it would be better to address this as a separate Issue. Because it won't work even if we fix just this part.